tests/integration/live_data/zookeeper_tests.py

```diff
@@ -48,3 +60,30 @@ def test_create_client_with_credentials(self):
 
         secrets.get_simple.assert_called_with("secret/zk-user")
         self.assertEqual(list(client.auth_data), [("digest", "myzkuser:hunter2")])
+
+    def test_create_client_uses_threading_handler_when_not_gevent_patched(self):
+        secrets = mock.Mock(spec=SecretsStore)
+        client = zookeeper_client_from_config(
+            secrets, {"zookeeper.hosts": "%s:%d" % zookeeper_endpoint.address}
+        )
+        assert isinstance(client.handler, SequentialThreadingHandler)
+
+    @mock.patch(
+        "kazoo.retry.KazooRetry.__init__.__defaults__",
+        (1, 0.1, 2, 0.4, 60.0, True, gevent.sleep, None, None),
```

Comment:
@nsheaps Here is the yucky patch. There are some tricks I could do with a context manager to only modify the one default arg, but I'm unsure if it's worth it.

Reply:
💡  If you're looking to unit test this, why not mock `baseplate.server.monkey.gevent_is_patched` and test which instance is in the handler without looking to actually monkey patch things?

Reply:
Unfortunately the Gevent handler throws a configuration error if the sleep function is unpatched, so we'd still need to patch `time.sleep` in this case.

```
kazoo.exceptions.ConfigurationError: Retry handler and event handler  must use the same sleep func
```

We could patch `gevent_is_patched` if we wanted to avoid doing any gevent patching in test though, that might make things cleaner.